satellite/referrals/service.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package referrals
+
+import (
+	"context"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/signing"
+	"storj.io/storj/pkg/storj"
+)
+
+var (
+	// Error is the default error class for referrals package.
+	Error = errs.Class("referrals")
+
+	// ErrReferralsConfigMissing is a error class for reporting missing referrals service configuration
+	ErrReferralsConfigMissing = errs.Class("misssing referrals service configuration")
+)
+
+type Config struct {
+	ReferralManagerURL storj.NodeURL
+}
+
+// ReferralsService allows communicating with the Referral Manager
+//
+// architecture: Service
+type Service struct {
+	log    *zap.Logger
+	signer signing.Signer
+	config Config
+	dailer rpc.Dialer
+}
+
+// NewService returns a service for handling referrals information.
+func NewService(log *zap.Logger, signer signing.Signer, config Config, dialer rpc.Dialer) *Service {
+	return &ReferralsService{
+		log:    log,
+		signer: signer,
+		config: config,
+		dailer: dialer,
+	}
+}
+
+func (service *Service) ReferralManagerConn(ctx context.Context) (*rpc.Conn, error) {
```

Comment:
Who is responsible for closing the connection? The caller? To me, it feels like the same place that does the dialing should close the connection.